### ISSUE TITLE
Add Firmware

### ISSUE THREAD

**Issue by Braypolk** [db-ok/SofleChocWireless#3]
Thanks for all the work you have done on this, I have just built this keyboard and it's absolutely awesome! I am having some trouble with firmware though.

**What feature would you like to see?**
I would like to see zmk firmware files also in this repo. An example that is specific to this keyboard and includes all the necessary files to get up and running.
Have a config that includes all features
- underglow rgb
-  nice!view and oled support
-  rotary support (looks like getting both to work can be tricky)

**Are you prepared to help?** 
From your pictures, it looks like you have a working build so I don't feel this is too big of an ask. But please let me know otherwise and I am happy to help. I could edit the readme/build guide to have an updated section for the firmware

**Comment by db-ok** [comment]
Glad to hear that you have built one of these! Thanks for trying it out!

Have you already looked through the Firmware and Programming section at the bottom of the build guide?

https://github.com/db-ok/SofleChocWireless/blob/main/docs/build_guide_choc_wireless.md#firmware-and-programming

That section of the build guide has some information about setting up your firmware build to support the nice!view as well as RGB underglow. OLED support is the default for the Sofle keyboard in ZMK, so you don't have to do anything special aside from enabling it in your .conf file with the following:
`CONFIG_ZMK_DISPLAY=y`

Rotary encoder support for the peripheral side is not something I have been able to set up for myself yet, as it has not been officially supported by the firmware in the past. I believe this is either in progress or could possibly even be complete as of right now, but I haven't built a new firmware recently (...the changes are very recent: https://zmk.dev/blog/2023/06/18/encoder-refactors). We should be getting official support for having both rotary encoders working soon, if it's not already done.

If the info in the build guide is not helpful or missing something about building your firmware, I would love to add to it or clarify what's there, or if you want to propose a change via PR, that would be great, too.

**Comment by db-ok** [comment]
I created a config repository to show examples of how to set up builds with the various combinations of features enabled: https://github.com/db-ok/zmk-config-soflechocwireless. I hope this will help get your build up and running @Braypolk

Each branch has a different combination of features enabled. Hopefully the branch names give a clear indication of which features are enabled, but I will be planning to update the build guide with a short explanation for each branch.

This repo contains branches for the combinations of features that work for the various ways I imagined that the board would be built, but there may be other combinations that folks want to build. These are just the ones I've tried out.
